src/main/java/com/cloudbees/jenkins/support/SupportAction.java

```diff
@@ -185,6 +181,78 @@ public void doDeleteBundles(StaplerRequest req, StaplerResponse rsp) throws Serv
         rsp.sendRedirect("");
     }
 
+    @RequirePOST
+    public void doDownloadBundles(StaplerRequest req, StaplerResponse rsp) throws ServletException, IOException {
+        JSONObject json = req.getSubmittedForm();
+        if (!json.has("bundles")) {
+            rsp.sendError(HttpServletResponse.SC_BAD_REQUEST);
+            return;
+        }
+
+        Set<String> bundlesToDownload = getSelectedBundles(req, json);
+        File fileToDownload = null;
+        if (bundlesToDownload.size() > 1) {
+            // more than one bundles were selected, create a zip file
+            fileToDownload = createZipFile(bundlesToDownload);
+        } else {
+            fileToDownload = new File(SupportPlugin.getRootDirectory(), bundlesToDownload.iterator().next());
+        }
+        logger.fine("Trying to download file "+ fileToDownload.getAbsolutePath());
+        try {
+            rsp.setContentType("application/zip");
+            rsp.addHeader("Content-Disposition", "inline; filename=" + fileToDownload.getName() + ";");
+            FileUtils.copyFile(fileToDownload, rsp.getOutputStream());
+        } catch (RuntimeException e) {
+            logger.log(Level.SEVERE, "Unable to download file " + fileToDownload.getAbsolutePath(), e);
+        } finally {
+            if (bundlesToDownload.size() > 1 && fileToDownload.delete()) {
+                logger.log(Level.FINE, "Temporary multiBundle file deleted: " + fileToDownload.getAbsolutePath());
+            } else {
+                logger.log(Level.SEVERE, "Unable to delete temporary multiBundle file: " + fileToDownload.getAbsolutePath());
```

Comment:
If there is a `RuntimeException` and `bundlesToDownload` is 1, then we will see `Unable to delete temporary multiBundle file: ...` that is wrong since there was not temporary bundle file created in that case.

IIUC, this shouldbe:

```
if (bundlesToDownload.size() > 1) {
    if (! fileToDownload.delete()) {
        logger.log(Level.FINE, "Temporary multiBundle file deleted: " + fileToDownload.getAbsolutePath());
    } else {
        logger.log(Level.SEVERE, "Unable to delete temporary multiBundle file: " + fileToDownload.getAbsolutePath());
    }
}
```

https://docs.oracle.com/javase/8/docs/api/java/io/File.html#createTempFile-java.lang.String-java.lang.String-